matrix/src/main/java/com/splunk/opentelemetry/appservers/javaee/HeaderDumpingServlet.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.appservers.javaee;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class HeaderDumpingServlet extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    StringWriter response = new StringWriter();
+    Enumeration<String> headerNames = req.getHeaderNames();
+    while (headerNames.hasMoreElements()) {
+      String headerName = headerNames.nextElement();
+      response.write(headerName + ": ");
+
+      List<String> headers = Collections.list(req.getHeaders(headerName));
+      if (headers.size() == 1) {
+        response.write(headers.get(0));
+      } else {
+        response.write("[");
+        for (String header : headers) {
+          response.write("  " + header + ",\n");
+        }
+        response.write("]");
+      }
+      response.write("\n");
+    }
+
+    ServletOutputStream outputStream = resp.getOutputStream();
+    outputStream.print(response.toString());
```

Comment:
You can write directly to `outputStream`, cannot you? No need for accumulator `StringWriter`